File: 2418-sort-the-people/2418-sort-the-people.cs
public class Solution {
    public string[] SortPeople(string[] names, int[] heights) {
        var n = names.Length;
        var dict = new Dictionary<int, string>();
        for(var i=0; i<n; i++)
            dict[heights[i]] = names[i];
        
        Array.Sort(heights, (h1, h2) => h2 - h1);
        for(var i=0; i < n; i++)
            names[i] = dict[heights[i]];
        
        return names;
    }
}

File: 2418-sort-the-people/2418-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        n = len(names)
        dic = {}
        for i in range(n):
            dic[heights[i]] = names[i]

        heights.sort(reverse=True)
        for i in range(n):
            names[i] = dic[heights[i]]

        return names
        